packages/instrumentation_adapter/lib/instrumentation_adapter.dart

```diff
@@ -32,14 +36,46 @@ class InstrumentationAdapterFlutterBinding
 
   static Map<String, String> _results = <String, String>{};
 
+  // Emulates the Flutter driver extension, returning 'pass' or 'fail'.
+  @override
+  void initServiceExtensions() {
+    super.initServiceExtensions();
+    Future<Map<String, dynamic>> callback(Map<String, String> params) async {
+      final String command = params['command'];
+      Map<String, String> response;
+      switch (command) {
+        case 'request_data':
+          final bool allTestsPassed = await _allTestsPassed.future;
+          response = <String, String>{
+            'message': allTestsPassed ? 'pass' : 'fail',
+          };
+          break;
+        case 'get_health':
+          response = <String, String>{'status': 'ok'};
+          break;
+        default:
+          throw UnimplementedError('$command is not implemented');
+      }
+      return <String, dynamic>{
+        'isError': false,
+        'response': response,
+      };
```

Comment:
When would this function return "isError": true?

Reply:
We have to attach `'isError': false` to the return value or the response will be rejected by `FlutterDriver`. I thought about returning `'isError': true` in the event of test failure.